src/pages/lock-exception/AddLockExceptionModal.js

```diff
@@ -0,0 +1,110 @@
+import { useAlert } from '@dhis2/app-runtime'
+import { useD2 } from '@dhis2/app-runtime-adapter-d2'
+import i18n from '@dhis2/d2-i18n'
+import { ButtonStrip, Button, CircularLoader } from '@dhis2/ui'
+import Dialog from 'material-ui/Dialog'
+import PropTypes from 'prop-types'
+import React, { useState } from 'react'
+import AddLockExceptionForm from './AddLockExceptionForm'
+
+const AddLockExceptionModal = ({
+    levels,
+    groups,
+    dataSets,
+    onAdd,
+    onClose,
+}) => {
+    const successAlert = useAlert(i18n.t('Lock exception added'), {
+        success: true,
+    })
+    const errorAlert = useAlert(
+        ({ error }) =>
+            i18n.t('Error adding lock exception: {{error}}', {
+                error,
+                nsSeparator: null,
+            }),
+        { critical: true }
+    )
+    const [isLoading, setIsLoading] = useState(false)
+    const [selectedOrgUnits, setSelectedOrgUnits] = useState([])
+    const [selectedDataSetId, setSelectedDataSetId] = useState(null)
+    const [selectedPeriodId, setSelectedPeriodId] = useState(null)
+    const { d2 } = useD2()
+    const handleAdd = async () => {
+        setIsLoading(true)
+        try {
+            await onAdd({
+                selectedOrgUnits,
+                selectedDataSetId,
+                selectedPeriodId,
+            })
+            successAlert.show()
+            onClose()
+        } catch (error) {
+            errorAlert.show({ error })
+        }
+        setIsLoading(false)
+    }
+    const isAddActionDisabled =
+        isLoading ||
+        !(selectedOrgUnits.length > 0 && selectedDataSetId && selectedPeriodId)
+
+    // TODO: Need to use material UI Dialog component instead of @dhis2/ui Modal
```

Comment:
I wonder if we could also use the @dhis2/ui Layer component to remedy this issue. It's meant to give control over the z-index, though I'm not sure if it would help here.

Reply:
The issue with the select inputs from material ui is that they render their dropdowns as popovers, so the z-index of the associated input component has no effect on their position relative to the modal. The best solution imo would be to replace all material ui components used in the data administration app with ones from `@dhis2/ui`, which would mean either reimplementing the organisation selection buttons from `d2-ui` or copy & pasting their code and changing their select field import from `material-ui` to `@dhis2/ui`